pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -0,0 +1,227 @@
+	v1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/internal/material"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	// pipelineConfigName is the name of the resolved dependency of the pipelineRef.
+	pipelineConfigName = "pipeline"
+	// taskConfigName is the name of the resolved dependency of the top level taskRef.
+	taskConfigName = "task"
+	// pipelineTaskConfigName is the name of the resolved dependency of the pipeline task.
+	pipelineTaskConfigName = "pipelineTask"
+	// inputResultName is the name of the resolved dependency generated from Type hinted ARTIFACT_INPUTS results
+	inputResultName = "inputs/result"
+	// pipelineResourceName is the name of the resolved dependency of pipeline resource.
+	pipelineResourceName = "pipelineResource"
+)
+
+// TaskRun constructs `predicate.resolvedDependencies` section by collecting all the artifacts that influence a taskrun such as source code repo and step&sidecar base images.
+func TaskRun(ctx context.Context, tro *objects.TaskRunObject) ([]*v1.ResourceDescriptor, error) {
+	var resolvedDependencies []*v1.ResourceDescriptor
+	var err error
+
+	// add top level task config
+	if p := tro.Status.Provenance; p != nil && p.RefSource != nil {
+		rd := v1.ResourceDescriptor{
+			Name:   taskConfigName,
+			URI:    p.RefSource.URI,
+			Digest: p.RefSource.Digest,
+		}
+		resolvedDependencies = append(resolvedDependencies, &rd)
+	}
+
+	mats := []common.ProvenanceMaterial{}
+
+	// add step images
+	if err := material.AddStepImagesToMaterials(tro.Status.Steps, &mats); err != nil {
+		return nil, err
+	}
+
+	// add sidecar images
+	if err := material.AddSidecarImagesToMaterials(tro.Status.Sidecars, &mats); err != nil {
+		return nil, err
+	}
+
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, convertMaterialsToResolvedDependencies(mats, "")...)
+
+	// add input artifacts
+	mats = artifacts.RetrieveMaterialsFromStructuredResults(ctx, tro, artifacts.ArtifactsInputsResultName)
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, convertMaterialsToResolvedDependencies(mats, inputResultName)...)
+
+	// add task resources
+	mats = material.AddTaskResourcesToMaterials(ctx, tro, []common.ProvenanceMaterial{})
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, convertMaterialsToResolvedDependencies(mats, pipelineResourceName)...)
+
+	// Store git rev as ResolvedDependencies
+	gitCommit, gitURL := material.GitInfo(tro)
+
+	if gitCommit != "" && gitURL != "" {
+		resolvedDependencies = append(resolvedDependencies, &v1.ResourceDescriptor{
+			Name:   inputResultName,
+			URI:    gitURL,
+			Digest: map[string]string{"sha1": gitCommit},
+		})
+	}
```

Comment:
Can we put the post processing steps into `GitInfo` as well so that we have a single place to process it and also be consistent with `AddTaskResourcesToMaterials`? This would also helps cleanup `Materials` function.